src-built-in/components/appLauncher2/src/components/LeftNav.jsx

```diff
@@ -0,0 +1,48 @@
+import React from  'react'
+import FoldersList from  './FoldersList'
+
+const bottomEntries = [
+	'New folder',
+	'New dashboard',
+	'App catalogue'
+]
+
+export default class LeftNav extends React.Component {
+
+	constructor(props) {
+		super(props)
+	}
+
+	render() {
+		return (
+			<div className="complex-menu-left-nav">
+			  <div className="top">
+				  {
+				  	this.props.folders.map((folder, index) => {
+				  		let className = 'complex-menu-section-toggle'
+				  		return (
+				  			<div onClick={() => this.props.onFolderClicked(folder)} 
+				  			className={className} key={index}>
+				  				{folder.name}
+				  			</div>
+				  			)
+				  	})
+				  }
+			  </div>
+			  <div className="bottom">
+				  {
+				  	bottomEntries.map((entry, index) => {
```

Comment:
There are two schools of thought here, and I don't know where I come down. Maybe @mesner has an opinion.

You can do what you're doing here. Inline array iteration spitting out DOM elements. Alternatively, you can break these out into functions, and just do this: `{this.renderBottomEntries()}`.

I'm not sure which is better but wanted to make you aware of both approaches. I tend to do what you're doing here.

Reply:
I didn't know about the second approach, but I really like it, I have moved the top links to a function and the bottom one to own component, the only reason I didn't break it into more components is that I don't know much about what the store looks like. so I will revisit structure at a later point.